tests/nvapi_d3d12.cpp

```diff
@@ -478,8 +496,10 @@ TEST_CASE("D3D12 methods succeed", "[.d3d12]") {
             params.pInfo = &info;
 
             D3D12_BUILD_RAYTRACING_ACCELERATION_STRUCTURE_INPUTS d3d12Desc{};
+            std::vector<D3D12_RAYTRACING_GEOMETRY_DESC> geometryDescs{};
             ALLOW_CALL(device, GetRaytracingAccelerationStructurePrebuildInfo(_, &info))
-                .LR_SIDE_EFFECT(d3d12Desc = *_1);
+                .LR_SIDE_EFFECT(d3d12Desc = *_1)
+                .LR_SIDE_EFFECT(if (_1->Type == D3D12_RAYTRACING_ACCELERATION_STRUCTURE_TYPE_BOTTOM_LEVEL && _1->DescsLayout == D3D12_ELEMENTS_LAYOUT_ARRAY) geometryDescs.assign(_1->pGeometryDescs, _1->pGeometryDescs + _1->NumDescs));
```

Comment:
I would combine both calls like:
```c++
.LR_SIDE_EFFECT({
    d3d12Desc = *_1;
    if (_1->Type == D3D12_RAYTRACING_ACCELERATION_STRUCTURE_TYPE_BOTTOM_LEVEL && _1->DescsLayout == D3D12_ELEMENTS_LAYOUT_ARRAY)
       geometryDescs.assign(_1->pGeometryDescs, _1->pGeometryDescs + _1->NumDescs);
});
```
Would also be cool to have a small comment that we cannot validate `pGeometryDescs` directly from `d3d12Desc` since is has already gone out of scope after the NVAPI method call..